src/plugin/fieldViews/CustomFieldView.ts

```diff
@@ -103,7 +103,14 @@ export class CustomFieldView<Value = unknown> implements FieldView<Value> {
   }
 
   public update(value: Value) {
-    this.updateOuterEditor(value);
+    // We must wrap our update call in a setTimeout to avoid a bug. If update
+    // may be called synchronously as the field is initially rendered by a
+    // consumer, we will be calling `updateOuterEditor` in the middle of the
+    // ProseMirror dispatch lifecycle, which can produce odd results. Running
+    // our update asynchronously avoids this issue.
+    setTimeout(() => {
+      this.updateOuterEditor(value);
+    });
```

Comment:
This makes sense as a solution given your thorough explanation. I have also tested this fix against a bug I found in Composer and found it to be fixed.